completions/ssh

```diff
@@ -427,6 +427,10 @@ _scp_path_esc='[][(){}<>"'"'"',:;^&!$=?`\\|[:space:]]'
 _scp_remote_files()
 {
     local IFS=$'\n'
+	
+    # Parse port number from whole command
+    __sshport=`echo ${COMP_WORDS[@]} | sed -n 's/^.*\( -\w*P\)\([ 0-9]*\)[ ].*/\2/p' `
+    [[ -n $"{__sshport// }" ]] && __sshoption="-p $__sshport" || __sshoption=""
```

Comment:
- `__sshport` should be made local. The variable name doesn't need to start with the double underscores in this function.
- `` `...` `` is deprecated by POSIX. `$(...)` should be used instead.
- `\w` is the GNU extension so is not always available.
- `[ 0-9]*` may unexpectedly capture multiple numbers separated by spaces.
- We call `sed` as `command sed` to prevent the alias of the same name from taking effect.
- `$"{...}"` must be a typo of `"${...}"`. Note that `$"..."` has a different meaning.

However, in the codebase of bash-completion, we usually scan all the words one by one looping over the array `words`. I'm afraid that the above code would capture some unexpected numbers in confusing filenames.

Reply:
Thank you for updating the PR! These are requests for further improvements and cosmetic changes:

- [ ] Could you quote the words as `"${COMP_WORDS[*]}"` to prevent unwanted pathname expansions?
- [ ] `\([ 0-9][0-9]*\}` can also match with the case without any digits. I would suggest `\( \{0,1\}[0-9]\{1,\}\)`. [ Note: `?` and `+` in typical regular-expression language are `\{0,1\}` and `\{1,\}`, respectively, in POSIX BRE. ]

The following ones are cosmetic/style changes.
- [ ] We'd like to remove `command` before `echo` for consistency with the codebase. I'm sorry that I should have made it clearer, but we actually have a rule that we prefix `command` *only* to the pre-defined set of commands that are commonly overridden by aliases (cf [here](https://github.com/scop/bash-completion/blob/0486a11ad5184d2bac8e6c988238e190a0d6cd03/test/runLint#L45) for the list). See also the corresponding section of [CONTRIBUTING.md](https://github.com/scop/bash-completion/blob/master/CONTRIBUTING.md) (where you can find the section by searching for the string `command sed`).
- [ ] Could you remove the quoting by `"..."` inside `[[ ... ]]`? In the conditional command `[[ ... ]]` the quoting is not needed because the pathname expansions and word splitting are not performed inside the conditional command. In the codebase, we are not quoting the parameter expansions inside the conditional command (except for the edge case of `${arr[*]}` where certain versions of Bash have a bug).
- [ ] Can you just use the form `[[ ${...} ]]` instead of `[[ -n ${...} ]]`? We are currently standardizing the choice of `[[ -n xxx ]]` vs `[[ xxx ]]` (see #740).

Reply:
> Thank you for updating the PR! These are requests for further improvements and cosmetic changes:
> 
> * [ ]  Could you quote the words as `"${COMP_WORDS[*]}"` to prevent unwanted pathname expansions?
> * [ ]  `\([ 0-9][0-9]*\}` can also match with the case without any digits. I would suggest `\( \{0,1\}[0-9]\{1,\}\)`. [ Note: `?` and `+` in typical regular-expression language are `\{0,1\}` and `\{1,\}`, respectively, in POSIX BRE. ]
> 
> The following ones are cosmetic/style changes.
> 
> * [ ]  We'd like to remove `command` before `echo` for consistency with the codebase. I'm sorry that I should have made it clearer, but we actually have a rule that we prefix `command` _only_ to the pre-defined set of commands that are commonly overridden by aliases (cf [here](https://github.com/scop/bash-completion/blob/0486a11ad5184d2bac8e6c988238e190a0d6cd03/test/runLint#L45) for the list). See also the corresponding section of [CONTRIBUTING.md](https://github.com/scop/bash-completion/blob/master/CONTRIBUTING.md) (where you can find the section by searching for the string `command sed`).
> * [ ]  Could you remove the quoting by `"..."` inside `[[ ... ]]`? In the conditional command `[[ ... ]]` the quoting is not needed because the pathname expansions and word splitting are not performed inside the conditional command. In the codebase, we are not quoting the parameter expansions inside the conditional command (except for the edge case of `${arr[*]}` where certain versions of Bash have a bug).
> * [ ]  Can you just use the form `[[ ${...} ]]` instead of `[[ -n ${...} ]]`? We are currently standardizing the choice of `[[ -n xxx ]]` vs `[[ xxx ]]` (see [style: prefer empty/`!` over `-n`/`-z` #740](https://github.com/scop/bash-completion/pull/740)).

The tests are still welcome if possible.